provisioner/terraform/resources.go

```diff
@@ -206,22 +219,37 @@ func ConvertState(modules []*tfjson.StateModule, rawGraph string) (*State, error
 			}
 
 			agent := &proto.Agent{
-				Name:                         tfResource.Name,
-				Id:                           attrs.ID,
-				Env:                          attrs.Env,
-				StartupScript:                attrs.StartupScript,
-				OperatingSystem:              attrs.OperatingSystem,
-				Architecture:                 attrs.Architecture,
-				Directory:                    attrs.Directory,
-				ConnectionTimeoutSeconds:     attrs.ConnectionTimeoutSeconds,
-				TroubleshootingUrl:           attrs.TroubleshootingURL,
-				MotdFile:                     attrs.MOTDFile,
-				StartupScriptBehavior:        startupScriptBehavior,
-				StartupScriptTimeoutSeconds:  attrs.StartupScriptTimeoutSeconds,
-				ShutdownScript:               attrs.ShutdownScript,
-				ShutdownScriptTimeoutSeconds: attrs.ShutdownScriptTimeoutSeconds,
-				Metadata:                     metadata,
-				DisplayApps:                  displayApps,
+				Name:                     tfResource.Name,
+				Id:                       attrs.ID,
+				Env:                      attrs.Env,
+				OperatingSystem:          attrs.OperatingSystem,
+				Architecture:             attrs.Architecture,
+				Directory:                attrs.Directory,
+				ConnectionTimeoutSeconds: attrs.ConnectionTimeoutSeconds,
+				TroubleshootingUrl:       attrs.TroubleshootingURL,
+				MotdFile:                 attrs.MOTDFile,
+				Metadata:                 metadata,
+				DisplayApps:              displayApps,
+			}
+			// Support the legacy script attributes in the agent!
+			if attrs.StartupScript != "" {
+				agent.Scripts = append(agent.Scripts, &proto.Script{
+					LogPath:          "coder-startup-script.log",
+					DisplayName:      "Startup Script",
+					Source:           attrs.StartupScript,
+					StartBlocksLogin: startupScriptBehavior == string(codersdk.WorkspaceAgentStartupScriptBehaviorBlocking),
+					Timeout:          attrs.StartupScriptTimeoutSeconds,
```

Comment:
We should delete these due to the new behavior (soft vs hard timeout), it's a bit of a lossy migration but I think that's better than forcibly killing the scripts that weren't before.